config.go

```diff
@@ -161,85 +160,123 @@ func (config TGFConfig) String() string {
 	return string(bytes)
 }
 
-func (config *TGFConfig) getAwsSession(duration int64) (*session.Session, error) {
-	if cachedSession != nil {
-		return cachedSession, nil
-	}
-	askedForMfa := false
-	options := awsSession.Options{
-		Profile:           config.tgf.AwsProfile,
-		SharedConfigState: awsSession.SharedConfigEnable,
-		AssumeRoleTokenProvider: func() (string, error) {
-			askedForMfa = true
-			fmt.Fprintf(os.Stderr, "Assume Role MFA token code: ")
-			v, err := terminal.ReadPassword(int(os.Stdin.Fd()))
-			fmt.Fprintln(os.Stderr)
-			return string(v), err
-		},
-	}
-	if duration > 0 {
-		options.AssumeRoleDuration = time.Duration(duration) * time.Second
+var cachedAwsConfig *aws.Config
+
+func (tgfConfig *TGFConfig) getAwsConfig(assumeRoleDuration time.Duration) (*aws.Config, error) {
+	if cachedAwsConfig != nil {
+		log.Debug("Using cached AWS config")
+		return cachedAwsConfig, nil
 	}
 
-	session, err := awsSession.NewSessionWithOptions(options)
+	log.Debugf("Creating new AWS config (assumeRoleDuration=%s)", assumeRoleDuration)
+	_config, err := awsConfig.LoadDefaultConfig(
+		context.TODO(),
+		awsConfig.WithSharedConfigProfile(tgfConfig.tgf.AwsProfile),
+		awsConfig.WithAssumeRoleCredentialOptions(func(o *stscreds.AssumeRoleOptions) {
+			o.TokenProvider = stscreds.StdinTokenProvider
+			if assumeRoleDuration > 0 {
+				o.Duration = assumeRoleDuration
+			}
+		}),
+	)
 
-	if err == nil {
-		// We must get the current credentials before verifying the expiration
-		_, err = session.Config.Credentials.Get()
+	if err != nil {
+		return nil, err
 	}
+	config := &_config
+
+	log.Debug("Fetching credentials for current AWS config")
+	creds, err := config.Credentials.Retrieve(context.TODO())
 	if err != nil {
-		return session, err
+		return config, nil
```

Comment:
Why `nil` and not `err` ? The error is sent higher and there is a condition on it.

Reply:
I was trying to match what the code did earlier but I clearly missed something. I though that the previous code ignored the error and just skipped ahead but I was wrong, it returns the error. I'll do that.

Reply:
I'm returning `nil, err` now since this is an error.